packages/@webex/plugin-cc/test/unit/spec/services/task/index.ts

```diff
@@ -0,0 +1,283 @@
+    expect(getErrorDetailsSpy).toHaveBeenCalledWith(error, 'accept', CC_FILE);
+  });
+
+  it('should decline call using webCallingService', async () => {
+    const declineCallSpy = jest.spyOn(webCallingService, 'declineCall');
+    const offSpy = jest.spyOn(webCallingService, 'off');
+    
+    await task.decline();
+
+    expect(declineCallSpy).toHaveBeenCalledWith(taskId);
+    expect(offSpy).toHaveBeenCalledWith(CALL_EVENT_KEYS.REMOTE_MEDIA, offSpy.mock.calls[0][1]);
+  });
+
+  it('should handle errors in decline method', async () => {
+    const error = {
+      details: {
+        trackingId: '1234',
+        data: {
+          reason: 'Decline Failed',
+        },
+      },
+    };
+
+    jest.spyOn(webCallingService, 'declineCall').mockImplementation(() => { throw error });
+    await expect(task.decline()).rejects.toThrow(new Error(error.details.data.reason));
+    expect(getErrorDetailsSpy).toHaveBeenCalledWith(error, 'decline', CC_FILE);
+  }); 
+
+  it('should hold the task', async () => {
+    await task.hold(taskId);
+
+    expect(contactMock.hold).toHaveBeenCalledWith({ interactionId: taskId, data: { mediaResourceId: taskDataMock.mediaResourceId } });
+  });
+
+  it('should handle errors in hold method', async () => {
+    const error = {
+      details: {
+        trackingId: '1234',
+        data: {
+          reason: 'Hold Failed',
+        },
+      },
+    };
+    contactMock.hold.mockImplementation(() => { throw error; });
+
+    await expect(task.hold(taskId)).rejects.toThrow(error.details.data.reason);
+    expect(getErrorDetailsSpy).toHaveBeenCalledWith(error, 'hold', CC_FILE);
+  });
+
+  it('should resume the task', async () => {
+    await task.resume(taskId);
+
+    expect(contactMock.unHold).toHaveBeenCalledWith({ interactionId: taskId, data: { mediaResourceId: taskDataMock.mediaResourceId } });
+  });
+
+  it('should handle errors in resume method', async () => {
+    const error = {
+      details: {
+        trackingId: '1234',
+        data: {
+          reason: 'Resume Failed',
+        },
+      },
+    };
+    contactMock.unHold.mockImplementation(() => { throw error; });
+
+    await expect(task.resume(taskId)).rejects.toThrow(error.details.data.reason);
+    expect(getErrorDetailsSpy).toHaveBeenCalledWith(error, 'resume', CC_FILE);
+  });
+
+  it('should end the task', async () => {
+    await task.end(taskId);
+
+    expect(contactMock.end).toHaveBeenCalledWith({ interactionId: taskId });
+  });
+
+  it('should handle errors in end method', async () => {
+    const error = {
+      details: {
+        trackingId: '1234',
+        data: {
+          reason: 'End Failed',
+        },
+      },
+    };
+    contactMock.end.mockImplementation(() => { throw error; });
+
+    await expect(task.end(taskId)).rejects.toThrow(error.details.data.reason);
+    expect(getErrorDetailsSpy).toHaveBeenCalledWith(error, 'end', CC_FILE);
+  });
+
+  it('should wrap up the task', async () => {
+    const wrapupPayload = {
```

Comment:
Few test scenarios that we should cover manually as well as in UT:

1. What happens if we try to wrapup without ending the call?
2. Will wrapup succeed if wrapuppayload received from the user is empty strings? How is our public method handling it?
3. Initiate a call, end it from the customer side and do not wrapup. Initiate another call and see if we are able to handle it

Reply:
I can answer for a few - 
2. It won't work... the string has to match the wrapup aux code name. It'll throw an error in that case.
3. I am not sure about this.... will check.